hawk_eye/data_generation/slice_image.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+""" Contains code to slice up an image into smaller tiles. """
+
+import argparse
+import pathlib
+from typing import List, Tuple
+
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+
+from hawk_eye.data_generation import generate_config as config
+
+
+def slice_image(
+    images: List, tile_size: Tuple[int, int], overlap: int, save_dir: pathlib.Path
+) -> None:
+    """ Take in an image and slice it into smaller images.
+    Args:
+        images: A list of paths to images to be sliced.
+        tile_size: The (width, height) of the tiles.
+        overlap: The overlap between adjacent tiles.
+        save_dir: The path to the directory within which to save slices.
+    Returns:
+        None.
+    """
+    for filename in tqdm(images, desc="Slicing images", total=len(images)):
+        image = Image.open(filename)
+        width, height = image.size
+
+        # Cropping logic repurposed from hawk_eye.inference.find_targets.tile_image()
+        for x in range(0, width, tile_size[0] - overlap):
+
+            # Shift back to extract tiles on the image
+            if x + tile_size[0] >= width and x != 0:
+                x = width - tile_size[0]
+
+            for y in range(0, height, tile_size[1] - overlap):
+
+                if y + tile_size[1] >= height and y != 0:
+                    y = height - tile_size[1]
+
+                tile = image.crop((x, y, x + tile_size[0], y + tile_size[1]))
+
+                tile.save(save_dir / f"{filename.stem}-{x}-{y}.JPG")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Slice an image into smaller images.")
+    parser.add_argument(
+        "--image_dir",
+        type=pathlib.Path,
+        required=False,
+        help="Path to directory of images to be sliced.",
+    )
+    parser.add_argument(
+        "--image_path",
+        type=pathlib.Path,
+        required=False,
+        help="Path to an image to be sliced.",
+    )
+    parser.add_argument(
+        "--image_extension",
+        type=str,
+        required=False,
+        default=".JPG",
+        help="Extension of images to slice.",
+    )
```

Comment:
```
   parser.add_argument(
        "--image_extensions",
        type=str,
        required=False,
        default=".jpg,.JPG",
        help="A comma-separated list of extensions of images to slice.",
    )
```

This is parsed by:
```
exts = [ext.strip() for ext in args.image_extensions.split(",")]
```

Reply:
Why do you want to do this AND set extension with `filename.suffix`?